tests/sources/xml/test_dspace_dim.py

```diff
@@ -216,3 +216,48 @@ def test_dspace_dim_transform_with_optional_fields_missing_transforms_correctly(
         format="electronic resource",
         content_type=["Not specified"],
     )
+
+
+def test_get_contents_success(dspace_dim_record_all_fields):
+    assert DspaceDim.get_contents(dspace_dim_record_all_fields) == ["Chapter 1"]
+
+
+def test_get_contents_transforms_correctly_if_fields_blank(
+    dspace_dim_record_optional_fields_blank,
+):
+    assert DspaceDim.get_contents(dspace_dim_record_optional_fields_blank) == []
+
+
+def test_get_contents_transforms_correctly_if_fields_missing(
+    dspace_dim_record_optional_fields_missing,
+):
+    assert DspaceDim.get_contents(dspace_dim_record_optional_fields_missing) == []
+
+
+def test_get_dates_success(dspace_dim_record_all_fields):
+    assert DspaceDim.get_dates(dspace_dim_record_all_fields, "abc123") == [
+        timdex.Date(kind="accessioned", value="2009-01-08T16:24:37Z"),
+        timdex.Date(kind="available", value="2009-01-08T16:24:37Z"),
+        timdex.Date(kind="Publication date", value="2002-11"),
+        timdex.Date(kind="coverage", note="1201-01-01 - 1965-12-21"),
+        timdex.Date(
+            kind="coverage",
+            range=timdex.DateRange(gte="1201-01-01", lte="1965-12-21"),
+        ),
+    ]
+
+
+def test_get_dates_transforms_correctly_if_fields_blank(
+    dspace_dim_record_optional_fields_blank,
+):
+    assert DspaceDim.get_dates(dspace_dim_record_optional_fields_blank, "abc123") == []
+
+
+def test_get_dates_transforms_correctly_if_fields_missing(
+    dspace_dim_record_optional_fields_missing,
+):
+    assert DspaceDim.get_dates(dspace_dim_record_optional_fields_missing, "abc123") == []
+
+
+def test_get_dates_invalid_date_range_skipped(dspace_dim_record_errors):
+    assert DspaceDim.get_dates(dspace_dim_record_errors, "abc123") == []
```

Comment:
In addition to `success`, `blank`, and `missing`; tests should be included for any edge cases handled differently by the code. This would vary depending on both the source and field